### PR TITLE
[DMD-768] Android Wallet Lock Timer (2)

### PR DESCRIPTION
**Pull request by tomasz-ludek**
The implementation was mainly taken from commit 9f9706df0bdfda500df92f80a8c797bae4ca061d,
merge conflicts were resolved and two issues fixed:
1. User was unable to unlock the wallet if he left the app in foreground before entering PIN during the very first launch.
2. App wasn't automatically locked when it remained foreground.

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/WalletActivity.java [hunk 1]
```diff
@@ -357,13 +367,8 @@ protected void error(final int messageResId, final Object... messageArgs) {
 
     @Override
     public boolean onCreateOptionsMenu(final Menu menu) {
-        super.onCreateOptionsMenu(menu);
-
         getMenuInflater().inflate(R.menu.wallet_options, menu);
-
-        MenuItem walletLockMenuItem = menu.findItem(R.id.wallet_options_lock);
-        walletLockMenuItem.setVisible(WalletLock.getInstance().isWalletLocked(wallet));
```
[the remark this answers is not in the document]
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/WalletActivity.java]: It should be handled by [AbstractWalletActivity.onOptionsItemSelected(MenuItem)](https://github.com/HashEngineering/dash-wallet/blob/wallet-lock-timer/wallet/src/de/schildbach/wallet/ui/AbstractWalletActivity.java#L105) after calling
`super.onCreateOptionsMenu(menu)`.
I will take a look.

File: wallet/testNet3/res/values/ic_launcher_background.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="ic_launcher_background">#FE8448</color>
+</resources>
```
[the remark this answers is not in the document]
Comment (tomasz-ludek) [reply wallet/testNet3/res/values/ic_launcher_background.xml]: Good point, I will move this fix to another PR.

[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/testNet3/res/values/ic_launcher_background.xml]: This doesn't need to be in another PR as it is already included in master.

File: wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java [hunk 3]
```diff
@@ -26,7 +30,9 @@
     private final static int RETRY_FAIL_TOLERANCE = 3;
     private final static int POW_LOCK_TIME_BASE = 6;
     private final static int FAIL_LIMIT = 8;
-    private final static long ONE_MINUTE_MILLIS = TimeUnit.MINUTES.toMillis(1);
+    public final static long THREE_MINUTE_MILLIS = TimeUnit.MINUTES.toMillis(3);
```
[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java]: @tomasz-ludek this still needs to be addressed.

File: wallet/src/de/schildbach/wallet/ui/WalletActivity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/WalletActivity.java]: @tomasz-ludek did you find a solution for this issue?

[the remark this answers is not in the document]
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/WalletActivity.java]: @HashEngineering this nasty bug was caused by the separation of showing/hiding the fiat value label and lock icon. The fiat value label visibility was managed by the WalletTransactionsFragment whereas the visibility of lock icon by AbstractWalletActivity.
Your screenshot shows the case when the wallet wasn't locked and after some time WalletTransactionsFragment received an event from TransactionsLoader. After that, it correctly updated the visibility of fiat value label (by hiding it) but didn't refresh the visibility of lock icon.

In my last changeset I moved most of the logic related to showing/hiding both controls into WalletTransactionsFragment and improve the code a bit. It should never happen again.

File: wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java [hunk 4]
```diff
@@ -70,6 +76,17 @@ public void successfulAttempt() {
         clearPreferences();
     }
 
+    /**
+     * When the PIN is entered successfully, save the time it was entered.
+     * @param config
+     */
+    public void successfulAttempt(Configuration config) {
+        long secureTime = System.currentTimeMillis();
```
[the remark this answers is not in the document]
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java]: I decided to get rid of PREFS_SECURE_TIME until we stabilize the wallet lock itself.
However, for the sake of convenience I added the potential implementation of this feature within https://github.com/HashEngineering/dash-wallet/pull/162/commits/78ebe7eedefa0a23edb1c7b3bba6b59f316369d9

File: wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java]: I changed this time back to ONE_MINUTE_MILLIS and added WalletLock.DEFAULT_LOCK_TIMER_MILLIS variable for the needs of wallet lock.

[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java]: Excellent.

File: wallet/src/de/schildbach/wallet/ui/WalletActivity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/WalletActivity.java]: Excellent.  This is no longer an issue.

**Comment by HashEngineering** [comment]
This PR will conflict with #201 in two places.

**Comment by tomasz-ludek** [comment]
_master_ merged into, all conflicts resolved.

**Comment by HashEngineering** [comment]
This will be merged manually.

**Comment by HashEngineering** [comment]
This was manually merged here:
https://github.com/HashEngineering/dash-wallet/commit/0b125e2e695e4fec30535eddefbac1a5d1543eee

**Pull request closed without merging**